src/shared_gui_components/EditView.cpp

```diff
@@ -400,4 +402,80 @@ void InputCheckBox::setIncomplete(bool incomplete) {
   }
 }
 
+PathInputView::PathInputView(const std::string& extension, bool isRead)
+  : InputView(), lineEdit(new QLineEdit()), nameLabel(new QLabel()), m_isRead(isRead) {
+  auto* vLayout = new QVBoxLayout();
+  vLayout->setContentsMargins(0, 0, 0, 0);
+  vLayout->setSpacing(5);
+  setLayout(vLayout);
+
+  nameLabel->setOpenExternalLinks(true);
+  nameLabel->setWordWrap(true);
+  vLayout->addWidget(nameLabel);
+
+  auto* hLayout = new QHBoxLayout();
+  hLayout->addWidget(lineEdit);
+
+  selectPathButton = new QPushButton("...");
+  selectPathButton->setFlat(true);
+  selectPathButton->setFixedSize(30, 20);
+  selectPathButton->setObjectName("StandardGrayButton");
+
+  connect(selectPathButton, &QPushButton::clicked, this, &PathInputView::onSelectPathButtonClicked);
+  hLayout->addWidget(selectPathButton);
+
+  vLayout->addLayout(hLayout);
+
+  // TODO: sanitize the input?
+  m_extension = toQString(extension);
+}
+
+void PathInputView::setName(const std::string& name, const boost::optional<std::string>& units, const boost::optional<std::string>& description) {
+  QString text;
+  text += QString::fromStdString(name);
+  if (units) {
+    text += QString::fromStdString(" (" + units.get() + ")");
+  }
+  if (description) {
+    text += QString::fromStdString("<div style=\"font-size:small;margin-top:2px;\">" + description.get() + "</div>");
+  }
+
+  nameLabel->setText(text);
+}
+
+void PathInputView::setIncomplete(bool incomplete) {
+  if (incomplete) {
+    nameLabel->setStyleSheet("QLabel { color: #DD0A05;}");
+  } else {
+    nameLabel->setStyleSheet("QLabel { color: black;}");
+  }
+}
+
+void PathInputView::setDisplayValue(const QVariant& value) {
+  lineEdit->setText(value.toString());
+}
+
+void PathInputView::onSelectPathButtonClicked() {
+  QString lastPath = lineEdit->text();
+
+  QString selectedPath;
+
+  if (m_extension.isEmpty()) {
+    // Assume this is a directory
+    selectedPath =
+      QFileDialog::getExistingDirectory(nullptr, tr("Open Directory"), lastPath, QFileDialog::ShowDirsOnly | QFileDialog::DontResolveSymlinks);
+  } else {
+    if (m_isRead) {
+      selectedPath = QFileDialog::getOpenFileName(nullptr, tr("Open Read File"), lastPath, m_extension);
+    } else {
+      selectedPath = QFileDialog::getSaveFileName(nullptr, tr("Select Save File"), lastPath, m_extension);
+    }
+  }
+
+  if (!selectedPath.isEmpty()) {
+    lineEdit->setText(selectedPath);  // QFileInfo(selectedPath).absoluteFilePath()
```

Comment:
Will selectedPath always be absolute?  What if the user wants it to be relative?  What would it be relative to?

Reply:
@jmarrec do you think the guidance should be that file path arguments are always absolute, used for reading/writing files outside of the workflow.  If there are files written by the measure, read by other measures, as part of the workflow those would just be found by file name (a string argument) using `WorkflowJSON::findFile` which searches the workflow paths?